### PR TITLE
Fixed #35882 -- Made migration questioner loop on all errors.

### PR DESCRIPTION
**Pull request by adamchainz**
#### Trac ticket number

ticket-35882

#### Branch description

Expand `InteractiveMigrationQuestioner._ask_default()` to loop on all exception types, update its display to show the exception type, and expand tests to existing types plus the `AttributeError` I encountered.

Thanks to @jacobtylerwalls for expanding the test coverage back in #15212, it let me make this change quite easily.

#### Checklist
- [x] This PR targets the `main` branch.
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [x] I have checked the "Has patch" ticket flag in the Trac system.
- [x] I have added or updated relevant tests.
- [x] I have added or updated relevant docs, including release notes if applicable.
- [x] I have attached screenshots in both light and dark modes for any UI changes.

### CODE REVIEW

**Review by charettes — COMMENTED** [review]
Any thoughts about adding another test ensures `KeyboardInterrupt` bubbles up without an explicit `exit`?

**Comment by adamchainz** [comment]
Good idea, done.

**Review by charettes — APPROVED** [review]

**Review by pauloxnet — APPROVED** [review]

File: tests/migrations/test_questioner.py [hunk 1]
```diff
@@ -61,10 +61,32 @@ def test_questioner_no_default_no_user_entry(self, mock_input):
...
+            self.questioner._ask_default()
+        self.assertIn(
+            "AttributeError: module 'datetime' has no attribute 'dat'",
+            self.prompt.getvalue(),
+        )
+
+    @mock.patch("builtins.input", side_effect=[KeyboardInterrupt()])
+    def test_questioner_no_default_keyboard_interrupt(self, mock_input):
+        with self.assertRaises(KeyboardInterrupt):
+            self.questioner._ask_default()
```
Comment (felixxm) [inline tests/migrations/test_questioner.py:89]: We could move this test to a separate commit since it's a test of existing behavior.

**Review by felixxm — APPROVED** [review]
@adamchainz Thanks :+1:
